src/components/number-input/number-input.vue

```diff
@@ -0,0 +1,247 @@
+<script setup lang="ts">
+import {
+  defineEmits,
+  defineAsyncComponent,
+  computed,
+  ref,
+  type Ref,
+  onBeforeMount,
+} from 'vue';
+import { v4 as uuidv4 } from 'uuid';
+
+
+type InputProps = {
+  label?: string
+  validationText?: string
+  isValid?: boolean | null
+  placeholder?: string | undefined
+  disabled?: boolean
+  modelValue?: number
+  helpertext?: string
+  maxlength?: number
+  maxvalue?: number
+  direction?: 'horizontal' | 'vertical'
+  increment?: number
+};
+
+const emit = defineEmits(['change', 'update:modelValue']);
+
+const props = withDefaults(defineProps<InputProps>(), {
+  validationText: 'validation text',
+  isValid: null,
+  placeholder: undefined,
+  label: '',
+  disabled: false,
+  helpertext: '',
+  modelValue: 0,
+  maxlength: 3,
+  maxvalue: 999,
+  direction: 'horizontal',
+  increment: 1,
+});
+
+const maxLengthRef: Ref = ref(props.maxlength);
+onBeforeMount(() => {
+  if (maxLengthRef.value > 14) maxLengthRef.value = 14;
+});
+
+
+function emitInput(e: HTMLInputElement)  {
+  if (e) {
+ const comparevalue = parseInt(e.value)
+  if ( comparevalue > props.maxvalue) {
+    e.value = props.maxvalue.toString();
+  }
+  if (e.value == '' ) {
+    e.value = '0';
+  }
+  e.value = e.value.slice(0, maxLengthRef.value);
+  const emitValue = parseInt(e.value.slice(0, maxLengthRef.value));
+
+  if (emitValue !== props.modelValue) {
+    emit('update:modelValue', emitValue);
+  }
+}
+}
+
+const verticalAlignment = props.direction == 'vertical' ? 'verticalStyle' : '';
+const labelRef = ref(props.label+ ' '+ uuidv4());
```

Comment:
We could go with only uuidV4 here, as laberef. I think that a space inside a `id` might mess things up. So only uuid will do just fine! :)

Reply:
removed label and space